reward/staking_manager.go

```diff
@@ -103,14 +103,29 @@ func GetStakingManager() *StakingManager {
 	return stakingManager
 }
 
-// GetStakingInfo returns a corresponding stakingInfo for a blockNum.
+// GetStakingInfo returns a stakingInfo on the staking block of the given block number.
+// Note that staking block is the block on which the associated staking information is stored and used during an interval.
 func GetStakingInfo(blockNum uint64) *StakingInfo {
+	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
+	return GetStakingInfoOnStakingBlock(stakingBlockNumber)
+}
+
+// GetStakingInfoOnStakingBlock returns a corresponding StakingInfo for a staking block number.
+// If the given number is not on the staking block, it returns nil.
+func GetStakingInfoOnStakingBlock(stakingBlockNumber uint64) *StakingInfo {
```

Comment:
What's the difference between `GetStakingInfoOnStakingBlock(sBN)` and `getStakingInfo(0, sBN)`?

Reply:
@blukat29 `GetStakingInfoOnStakedBlock` is introduced here to differentiate `GetStakingInfo` without calling `CalcStakingBlockNumber`. As you see, it checks `stakingManager`is nil and it is public method. Any suggestion?

Reply:
@jeongkyun-oh The nil check is also in `getStakingInfo`. So `GetStakingInfoOnStakedBlock` is simply a wrapper that adds the '0' argument.

I'd say we can only leave the public method unless there are other uses of `getStakingInfo`.

Reply:
@blukat29 I see. I have updated. Thanks :)